nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/lossfunctions/SDLossMAE.java

```diff
@@ -0,0 +1,17 @@
+package org.nd4j.linalg.lossfunctions;
+
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+
+public class SDLossMAE extends SameDiffLoss {
+    public SDLossMAE(){
+        super();
+    }
+
+    @Override
+    public SDVariable defineLoss(SameDiff sd, SDVariable layerInput, SDVariable labels) {
+
+        return sd.math.abs(labels.sub(sd.getVariable("out"))).mean(1);
```

Comment:
Don't use sd.getVariable("out") - there's no such variable. Just use `layerInput` instead.

Same thing in SDLossMSE